### PR TITLE
KAFKA-14683 Cleanup WorkerSinkTaskTest

### PR DESCRIPTION
**Pull request by hgeraldino**
Follow up of https://github.com/apache/kafka/pull/15316

* Rename `WorkerSinkTaskMockitoTest` back to `WorkerSinkTaskTest`
* Tidy up the code a bit

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Review by chia7712 — COMMENTED** [review]
@hgeraldino thank for this patch. Could you please apply `assertThrow` to this class? For example, the following assert can be rewritten.
*origin*
```java
        try {
            workerTask.iteration();
            fail("Poll should have raised the rebalance exception");
        } catch (RuntimeException e) {
            assertEquals(exception, e);
        }
```
*new*
```java
assertEquals(exception, assertThrows(RuntimeException.class, () -> workerTask.iteration()));
```

**Comment by hgeraldino** [comment]
> @hgeraldino thank for this patch. Could you please apply `assertThrow` to this class? For example, the following assert can be rewritten. _origin_
> 
> ```java
>         try {
>             workerTask.iteration();
>             fail("Poll should have raised the rebalance exception");
>         } catch (RuntimeException e) {
>             assertEquals(exception, e);
>         }
> ```
> 
> _new_
> 
> ```java
> assertEquals(exception, assertThrows(RuntimeException.class, () -> workerTask.iteration()));
> ```

Sure thing! 
Done

**Review by chia7712 — APPROVED** [review]
LGTM

**Comment by hgeraldino** [comment]
Thanks @chia7712 for your review

@gharris1727 does this looks good to you? It's the final PR to consider the `WorkerSinkTaskTest` migration done done

File: connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerSinkTaskTest.java [hunk 1]
```diff
@@ -553,12 +547,8 @@ public void testErrorInRebalancePartitionLoss() {
         workerTask.iteration();
         verifyPollInitialAssignment();
 
-        try {
-            workerTask.iteration();
-            fail("Poll should have raised the rebalance exception");
-        } catch (RuntimeException e) {
-            assertEquals(exception, e);
-        }
+        Throwable thrownException = assertThrows(RuntimeException.class, () -> workerTask.iteration());
```
Comment (OmniaGM) [inline connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerSinkTaskTest.java]: The type of the `thrownException` should be `RuntimeException` as it has been casted when we called `assertThrows(RuntimeException.class, <lambda>)`

File: connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerSinkTaskTest.java [hunk 2]
```diff
@@ -580,12 +570,9 @@ public void testErrorInRebalancePartitionRevocation() {
...
         workerTask.iteration();
         verifyPollInitialAssignment();
-        try {
-            workerTask.iteration();
-            fail("Poll should have raised the rebalance exception");
-        } catch (RuntimeException e) {
-            assertEquals(exception, e);
-        }
+
+        Throwable thrownException = assertThrows(RuntimeException.class, () -> workerTask.iteration());
```
Comment (OmniaGM) [inline connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerSinkTaskTest.java]: same, so am not going to point it out for each test.

File: connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerSinkTaskTest.java [hunk 3]
```diff
@@ -1215,17 +1195,13 @@ public void testSuppressCloseErrors() {
...
-            workerTask.execute();
-            fail("workerTask.execute should have thrown an exception");
-        } catch (ConnectException e) {
-            assertSame("Exception from put should be the cause", putException, e.getCause());
-            assertTrue("Exception from close should be suppressed", e.getSuppressed().length > 0);
-            assertSame(closeException, e.getSuppressed()[0]);
-        }
+
+        Throwable thrownException = assertThrows(ConnectException.class, () -> workerTask.execute());
+        assertSame("Exception from put should be the cause", putException, thrownException.getCause());
```
Comment (OmniaGM) [inline connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerSinkTaskTest.java]: While `assertSame` and `assertEquals` behave the same when the default equals implementation is provided (which compare references)  but I'm curious why are we opting in for `assertSame` over `assertEquals`. Do we need the test to explicitly comparing object reference or just value?

Comment (hgeraldino) [reply connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerSinkTaskTest.java]: This is a remnant of the original implementation. As you correctly guessed, there is no difference in this case between using `assertEquals` and `assertSame`: neither the `RuntimeException` class nor any of its parents override the `equals()` method, so they both are effectively checking for reference equality (as implemented by the base `Object` class)

I'm happy to change these to `assertEquals` (this is the only place where this is used)

**Comment by hgeraldino** [comment]
Thanks @OmniaGM for your review

File: connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerSinkTaskTest.java (hunk 3 above)
Comment (OmniaGM) [reply connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerSinkTaskTest.java]: For me it is more about the readability. When I see `assertSame` it makes me think we are comparing object ref while `assertEquals` for me means we just care about executing `equal` method. I personally would lean toward using `assertEquals`

Comment (gharris1727) [reply connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerSinkTaskTest.java]: When I wrote this test originally, I did intentionally use assertSame to do a reference equality check.

Comment (OmniaGM) [reply connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerSinkTaskTest.java]: I can't see why we care about the ref in some of these tests but if this is the intent then keeping `assertSame` is okay. My comment was more a question whether we need the test to be explicit or not.

Comment (chia7712) [reply connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerSinkTaskTest.java]: @OmniaGM @gharris1727 I'd like to ship it tomorrow if both of you have no objection.

Comment (OmniaGM) [reply connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerSinkTaskTest.java]: LGTM

**Review by OmniaGM — APPROVED** [review]

**Pull request merged**